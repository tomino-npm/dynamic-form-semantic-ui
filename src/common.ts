import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: (...params: any[]) => any };
};

export const { css, renderCss } = initCss('d-form');
