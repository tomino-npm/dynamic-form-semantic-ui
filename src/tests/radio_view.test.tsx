import * as React from 'react';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

const schema: JSONSchema = {
  type: 'object',
  properties: {
    religion: {
      type: 'string'
    },
    lined_religion: {
      type: 'string'
    },
    religions: {
      enum: [
        { text: 'Christian', value: 'CH' },
        { text: 'Buddhist', value: 'BU' },
        { text: 'Jedi', value: 'JE' }
      ]
    }
  }
};

const controlData = { religion: 'JE', lined_religion: 'BU' };

const formDefinition: FormDefinition = create.form({
  elements: [
    create.formElement({
      row: 0,
      column: 0,
      width: 16,
      list: 'religions',
      control: 'Radio',
      label: 'Religions',
      source: 'religion',
      inline: true
    }),
    create.formElement({
      row: 1,
      column: 0,
      width: 16,
      list: 'religions',
      control: 'Radio',
      label: 'Religions',
      source: 'lined_religion',
      vertical: true
    })
  ]
});

const form = new FormModel(formDefinition, schema, controlData);

// just another notation
describe('Radio', () => {
  return {
    componentWithData: () => <TestComponent form={form} />
  };
});
