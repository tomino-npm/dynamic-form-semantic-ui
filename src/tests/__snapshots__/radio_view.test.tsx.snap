// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Radio changes values 1`] = `
<div
  className="ui segment ui form"
>
  <style>
    .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
  </style>
  <div
    className="ui form"
  >
    <div
      className="fields"
    >
      <div
        className="inline field"
      >
        <label
          htmlFor="religion"
        >
          Religions
        </label>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="CH"
          />
          <label>
            Christian
          </label>
        </div>
        <div
          className="ui checked radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="BU"
          />
          <label>
            Buddhist
          </label>
        </div>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="JE"
          />
          <label>
            Jedi
          </label>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="lined_religion"
        >
          Religions
        </label>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="CH"
          />
          <label>
            Christian
          </label>
        </div>
        <br />
        <div
          className="ui checked radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="BU"
          />
          <label>
            Buddhist
          </label>
        </div>
        <br />
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="JE"
          />
          <label>
            Jedi
          </label>
        </div>
        <br />
      </div>
    </div>
  </div>
</div>
`;

exports[`Form Radio renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <style>
    .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
  </style>
  <div
    className="ui form"
  >
    <div
      className="fields"
    >
      <div
        className="inline field"
      >
        <label
          htmlFor="religion"
        >
          Religions
        </label>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="CH"
          />
          <label>
            Christian
          </label>
        </div>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="BU"
          />
          <label>
            Buddhist
          </label>
        </div>
        <div
          className="ui checked radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name="religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="JE"
          />
          <label>
            Jedi
          </label>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="lined_religion"
        >
          Religions
        </label>
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="CH"
          />
          <label>
            Christian
          </label>
        </div>
        <br />
        <div
          className="ui checked radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="BU"
          />
          <label>
            Buddhist
          </label>
        </div>
        <br />
        <div
          className="ui radio checkbox d-form-Adiagibjfa"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name="lined_religion"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value="JE"
          />
          <label>
            Jedi
          </label>
        </div>
        <br />
      </div>
    </div>
  </div>
</div>
`;
